codegen/src/main/twirl/templates/PlayScalaServer/Router.scala.txt

```diff
@@ -0,0 +1,49 @@
+@*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ *@
+
+@(service: akka.grpc.gen.scaladsl.Service)
+
+package @service.packageName
+
+import javax.inject._
+
+import scala.concurrent.Future
+
+import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
+import akka.stream.Materializer
+
+import play.api.mvc._
+import play.api.mvc.akkahttp.AkkaHttpHandler
+import play.api.routing.Router
+import play.api.routing.Router.Routes
+
+@@Singleton
+class @{service.name}Router @@Inject()(impl: controllers.@{service.name}Impl)(implicit mat: Materializer)
+  extends Router {
+
+  protected val prefix: String = @{service.name}.name
```

Comment:
I'm not sure, but I think the prefix might be set by the parent router based on the path given in the `routes` file.

Reply:
Yes. Given:
* Play will create the route and call `withPrefix` on it with the prefix from the `routes` file
* the gRPC is opinionated on using `@{service.name}.name` as the prefix ("[Some gRPC implementations may allow the Path format shown above to be overridden, but this functionality is strongly discouraged](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md)").

in zoom @renatocaval and I figured the most reasonable behavior would be to default to the prefix encouraged by the spec, and tell Play users to use the name in the `routes` file (#257) but honor a custom prefix when they configure it.

I now realize that this implementation doesn't quite achieve that yet, I'll add tests to both document and check the behavior ;).